packages/changeset-bot/index.test.js

```diff
@@ -0,0 +1,154 @@
+const nock = require('nock');
+const { Probot } = require('probot');
+const changesetBot = require('.');
+const outdent = require('outdent');
+
+const pullRequestOpen = require('./test/fixtures/pull_request.opened');
+const pullRequestSynchronize = require('./test/fixtures/pull_request.synchronize');
+
+nock.disableNetConnect();
+
+describe('changeset-bot', () => {
+  let probot;
+
+  beforeEach(() => {
+    probot = new Probot({});
+    const app = probot.load(changesetBot);
+
+    // just return a test token
+    app.app = () => 'test';
+  });
+
+  it('should add a comment when there is no comment', async () => {
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/issues/1/comments')
+      .reply(200, []);
+
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/pulls/1/files')
+      .reply(200, [
+        { filename: '.changeset/something/changes.md', status: 'added' },
+      ]);
+
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/pulls/1/commits')
+      .reply(200, [{ sha: 'ABCDE' }]);
+
+    nock('https://api.github.com')
+      .post('/repos/pyu/testing-things/issues/1/comments', body => {
+        expect(body.comment_id).toBeNull();
+        return true;
+      })
+      .reply(200);
+
+    await probot.receive({
+      name: 'pull_request',
+      payload: pullRequestOpen,
+    });
+  });
+
+  it('should update a comment when there is a comment', async () => {
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/issues/1/comments')
+      .reply(200, [
+        {
+          id: 7,
+          user: {
+            login: 'changeset-bot[bot]',
+          },
+        },
+      ]);
+
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/pulls/1/files')
+      .reply(200, [
+        { filename: '.changeset/something/changes.md', status: 'added' },
+      ]);
+
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/pulls/1/commits')
+      .reply(200, [{ sha: 'ABCDE' }]);
+
+    nock('https://api.github.com')
+      .patch('/repos/pyu/testing-things/issues/comments/7', body => {
+        expect(body.number).toBe(1);
```

Comment:
Quick q: Why is the shape of this test (testing the body.number) different to the shape of the test above it (testing comment_id)? I'd be expecting these to test the same property.

Reply:
Update requests aren't posted by issue/PR number like new comments are. Instead they are done across all comments in the repo. The comment id is included at the end of the endpoint path, so the request body just includes the issue/PR number.